crates/swc_ecma_transforms_module/src/util.rs

```diff
@@ -59,11 +59,19 @@ const fn default_strict_mode() -> bool {
     true
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct LazyObjectConfig {
+    #[serde(with = "serde_regex")]
+    pub patterns: Vec<regex::Regex>,
```

Comment:
@kdy1 is this what you had in mind?

Reply:
Oh, sorry seems like I missed the email.

I think we should accept `Vec<JsWord>` and cache the regex based on the given string.

Reply:
Just making sure: you mean `patterns: Vec<JsWord>` instead of `patterns: Vec<regex::Regex>`? And then cache the transformation from `JsWord` to `Regex`?

If that's what you mean, are you fine with changing the failure mode from parse-time to run-time?

Reply:
Yes, but we don't need to change it to parse time.
I didn't get what you mean.

We create modules pass anyway (at parse time), and we can pass cached regex to it.

Reply:
What I meant is that right now, an error in the regex pattern in the `.swcrc` file would be caught at parse-time, when the file is parsed, because it's using `serde_regex`. If we allow `JsWord`in the config, and transform the `JsWord` into a regex while evaluating `is_lazy()`, it's switching the error to a different phase, at run-time.

> We create modules pass anyway (at parse time), and we can pass cached regex to it.

Oh! Can you show me how/where to do that? I thought your proposition was to replace it with something like:

```rust
pub struct LazyObjectConfig {
    patterns: Vec<JsWord>,
    regexes: Option<Vec<Regex>>,
}

impl LazyObjectConfig {
    // not sure exactly how to do this because is_lazy() doesn't receive `mut &self`, just `&self`.
    pub fn matches(&mut self, path: &str) -> Result<bool> {
        let regexes = self.regexes.get_or_insert_with(|| {
            let r = vec![];
            for pat in self.patterns {
                r.push(Regex::new(pat)?);
            }
            r
        })?;
    }
}
```

Reply:
No, it's not runtime vs parse time.
We have to convert it to `Arc<Regex>>`, because we have to pass it to module pass like common js.
I think you mean passing it to `common_js` pass as a `Vec<JsWord>`, but that's not what I mean.

Reply:
I don't follow what you're trying to say then. Where would you convert from `Vec<JsWord` to `Arc<Regex>`?

I'm sorry, I really don't have the full overview of the project, so I may be missing some important parts.

Reply:
While creating the config instance for module transform.

https://docs.rs/swc_ecma_transforms_module/0.84.2/swc_ecma_transforms_module/util/struct.Config.html

Especially, the `Lazy` type should be extended to accept `Vec<Arc<Regex>>` variant.